Sources/ProcedureKit/Procedure.swift

```diff
@@ -151,25 +198,52 @@ open class Procedure: Operation, ProcedureProtocol {
     /// handle responding to cancellation as appropriate.
     ///
     final public override var isCancelled: Bool {
-        return _stateLock.withCriticalScope { _isCancelled }
+        return stateLock.withCriticalScope { _isCancelled }
+    }
+
+    /// The Boolean indicators to be used *within* the stateLock
+    private var _isEnqueued: Bool {
+        return _state >= .pending
+    }
+    private var _isPending: Bool {
+        return _state == .pending
+    }
+    private var _isExecuting: Bool {
+        return _state == .executing
+    }
+    private var _isFinished: Bool {
+        return _state == .finished
     }
 
     // MARK: Protected Internal Properties
 
-    fileprivate struct ProtectedProperties {
+    // Grouped in a class to allow for easily deinitializing in `deinit`.
+    fileprivate class ProtectedProperties {
         var log: LoggerProtocol = Logger()
         var errors = [Error]()
         var observers = [AnyObserver<Procedure>]()
         var directDependencies = Set<Operation>()
         var conditions = Set<Condition>()
     }
-    fileprivate let protectedProperties = Protector(ProtectedProperties())
+    fileprivate var protectedProperties: ProtectedProperties! = ProtectedProperties()
```

Comment:
To permit easy (and explicit) deinit in `Procedure.deinit` (inside the lock, to resolve a data race).